lib/glossarist/concept.rb

```diff
@@ -1,244 +1,101 @@
-    attr_reader :id
-
-    attr_writer :uuid
-
-    # Concept designations.
-    # @todo Alias +terms+ exists only for legacy reasons and will be removed.
-    # @return [Array<Designations::Base>]
-    attr_reader :designations
-    alias :terms :designations
-
-    # <<BasicDocument>>LocalizedString
-    # @return [String]
-    attr_accessor :domain
-
-    # <<BasicDocument>>LocalizedString
-    # @return [String]
-    attr_accessor :subject
-
-    # Concept definition.
-    # @return [Array<DetailedDefinition>]
-    attr_reader :definition
-
-    # Non verbal representation of the concept.
-    # @return [NonVerbRep]
-    attr_accessor :non_verb_rep
-
-    # Concept notes
-    # @return [Array<DetailedDefinition>]
-    attr_reader :notes
-
-    # Concept examples
-    # @return [Array<DetailedDefinition>]
-    attr_reader :examples
-
-    # Contains list of extended attributes
-    attr_accessor :extension_attributes
-
-    attr_accessor :lineage_source
-    attr_accessor :lineage_source_similarity
-
-    attr_accessor :release
-
-    def initialize(*args)
-      @localizations = {}
-      @sources = Glossarist::Collections::Collection.new(klass: ConceptSource)
-      @related = Glossarist::Collections::Collection.new(klass: RelatedConcept)
-      @definition = Glossarist::Collections::Collection.new(klass: DetailedDefinition)
-      @notes = Glossarist::Collections::Collection.new(klass: DetailedDefinition)
-      @examples = Glossarist::Collections::Collection.new(klass: DetailedDefinition)
-      @dates = Glossarist::Collections::Collection.new(klass: ConceptDate)
-
-      @designations = Glossarist::Collections::DesignationCollection.new
-      @extension_attributes = {}
-
-      normalize_args(args)
-
-      super
+  class Concept < Lutaml::Model::Serializable
+    attribute :data, ConceptData, default: -> { ConceptData.new }
+    attribute :id, :string
+    attribute :uuid, :string
+    attribute :subject, :string
+    attribute :non_verb_rep, :string
+    attribute :extension_attributes, :string
+    attribute :lineage_source, :string
+    attribute :localizations, :hash
+    attribute :extension_attributes, :hash
+    attribute :termid, :string
+
+    yaml do
+      map :data, to: :data
+      map :termid, to: :termid
+      map :subject, to: :subject
+      map :non_verb_rep, to: :non_verb_rep
+      map :extension_attributes, to: :extension_attributes
+      map :lineage_source, to: :lineage_source
+      map :localizations, to: :localizations
+      map :extension_attributes, to: :extension_attributes
+
+      map :date_accepted, with: { from: :date_accepted_from_yaml, to: :date_accepted_to_yaml }
+      map :uuid, to: :uuid, with: { to: :uuid_to_hash, from: :uuid_from_hash }
+      map :id, to: :id, with: { to: :id_to_hash, from: :id_from_hash }
+      map :identifier, to: :id, with: { to: :id_to_hash, from: :id_from_hash }
+    end
+
+    def designations
+      data.terms
     end
+    alias :terms :designations
```

Comment:
Methods like this are just to call the delegated methods. 

May be we should add a `delegate` key with `attribute` in `lutaml-model` to add methods like this i.e

```ruby
class Concept < Lutaml::Model::Serializable
  attribute :terms, delegate: :data
end
```

@ronaldtse What do you suggest?